docs/TUTORIAL.md

```diff
@@ -1,3 +1,305 @@
+exercises, you create an explicit IngressController and Service from the nginx 
+deployment described in the [readme]. You simplify the 
+process of defining and adding these new objects by taking advantage of 
+**ksonnet**'s built-in mixins.
+
+First, define base configuration values, dependent objects, and the IngressController:
+
+```javascript
+local k = import "ksonnet.beta.2/k.libsonnet";
+
+// specify the necessary import objects
+local deployment = k.extensions.v1beta1.deployment;
+local container = deployment.mixin.spec.template.spec.containersType;
+
+// base configuration. These values can be passed explicitly instead, but this 
+// approach lets you reuse values for greater efficiency.
+local conf = {
+    namespace: "nginx-ingress",
+    ingressController: {
+        name: "nginx-ingress-controller",
+        selector: {"run": $.ingressController.name},
+        replicas: 2,
+        terminationGracePeriodSeconds: 60,
+        container: {
+            image: "gcr.io/google_containers/nginx-ingress-controller:0.9.0-beta.7",
+            probe: {
+                path: "/healthz",
+                port: 10254,
+                initialDelaySeconds: 10,
+                timeoutSeconds: 1,
+                scheme: "HTTP",
+            },
+            ports: {
+                http: 80,
+                https: 443,
+            },
+        },
+    },
+};
+
+// define the metadata that the IngressController object needs
+local metadata = deployment.mixin.metadataType;
+local controllerMetadata = metadata.new() +
+    metadata.name(conf.ingressController.name) +
+    metadata.labels(conf.ingressController.selector) +
+    metadata.namespace(conf.namespace);
+
+// define the container for the deployment
+local nginxContainer = container.new(conf.ingressController.name, conf.ingressController.container.image) +
+    container.ports([
+        container.portsType.containerPort(conf.ingressController.container.ports.http) +
+        container.portsType.hostPort(conf.ingressController.container.ports.http),
+
+        container.portsType.containerPort(conf.ingressController.container.ports.https) +
+        container.portsType.hostPort(conf.ingressController.container.ports.https),
+    ]);
+
+// define the deployment
+local ingressController = deployment.new(conf.ingressController.name, conf.ingressController.replicas, [nginxContainer], conf.ingressController.selector) +
+    deployment.mixin.metadata.mixinInstance(controllerMetadata) +
+    deployment.mixin.spec.template.metadata.annotations(conf.ingressController.annotations) +
+    deployment.mixin.spec.template.spec.terminationGracePeriodSeconds(conf.ingressController.terminationGracePeriodSeconds);
+
+// call the deployment
+k.core.v1.list.new(ingressController)
+```
+
+Save the file as `ingressController.jsonnet` at the root of your **ksonnet** fork,
+and run:
+
+```bash
+jsonnet ingressController.jsonnet > deployment.yaml
+kubectl apply -f deployment.yaml
```

Comment:
can you move the `kubectl` command to *after* the part where you show the generated YAML? Since deploying to the cluster itself feels like a different step.